lib/index.js

```diff
@@ -277,6 +280,24 @@ function nonEmptyArray(speck) {
     fc.array(speck._fastCheckArbitrary, 1, 5));
 }
 
+/**
+ * The speck equivalent of the TypeScript `Record<..>` type.
+ *
+ * e.g. `s.record(s.string, s.float)` would have TS type `Record<string, number>`.
+ *
+ * @template {typeof string} TKeySpeck
+ * @template {import('./types').Speck<any>} TValueSpeck
+ * @param {TKeySpeck} keySpeck
+ * @param {TValueSpeck} valueSpeck
+ * @returns {import('./types').RecordType<TKeySpeck, TValueSpeck>}
+ */
+function record(keySpeck, valueSpeck) {
+  // TODO fix the type here
+  return /** @type {any} */(createObjectSpeck(
```

Comment:
I am too lazy rn to fix this, but I know that the type is right. You can trust me though 🙃

Reply:
Isn't this just @type `{Record<TKeySpec, TValueSpec>}` ? something or other

Reply:
The problem is that `io-ts` is not giving the right type. Rather than using the generic types in TKeySpeck and TValueSpeck, it is returning something like `io-ts.Record<string, any>`.

TypeScript will then rightly complain that this function's return value (which will be inferred from io-ts to become `RecordType<string, any>`) is not specific enough to match the signature, which is `RecordType<TypeOf<TKeySpeck>, TypeOf<TValueSpeck>>`

Reply:
@Mortuie The problem is that `io-ts` is not giving the right type. Rather than using the generic types in TKeySpeck and TValueSpeck, it is returning something like `io-ts.Record<string, any>`.

TypeScript will then rightly complain that this function's return value (which will be inferred from io-ts to become `RecordType<string, any>`) is not specific enough to match the signature, which is `RecordType<TypeOf<TKeySpeck>, TypeOf<TValueSpeck>>`

Reply:
Casting something to `any` is just a way to get TypeScript to completely ignore an error for a particular expression